src/PreconditionerTypes.f90

```diff
@@ -343,6 +550,695 @@ SUBROUTINE setup_ILU_PreCondtype(thisPC)
     ENDIF
   ENDIF
 ENDSUBROUTINE setup_ILU_PreCondtype
+!
+!-------------------------------------------------------------------------------
+!> @brief Intializes up the SOR Preconditioner Type with a parameter list
+!> @param thisPC The preconditioner to act on
+!> @param A The matrix to precondition
+!> @param params The parameter list
+!>
+SUBROUTINE init_SOR_PreCondtype(thisPC,A,params)
+  CHARACTER(LEN=*),PARAMETER :: myName='init_RSOR_PreCondType'
+  CLASS(SOR_PrecondType),INTENT(INOUT) :: thisPC
+  CLASS(MatrixType),ALLOCATABLE,TARGET,INTENT(IN),OPTIONAL :: A
+  TYPE(ParamType),INTENT(IN),OPTIONAL :: params
+  TYPE(ParamType)::PListMat_LU
+  INTEGER(SIK)::k
+
+  !required statuses
+  REQUIRE(.NOT. thisPC%isinit)
+  REQUIRE(PRESENT(A))
+  REQUIRE(ALLOCATED(A))
+  REQUIRE(A%isInit)
+
+  thisPC%A => A
+
+  !gets the number of blocks from the parameter list
+  CALL params%get('PCType->numBlocks',thisPC%numBlocks)
+  CALL params%get('PCType->omega',thisPC%omega)
+
+  !makes sure that the number of blocks is valid
+  REQUIRE(thisPC%numBlocks .GT. 0)
+
+  !makes sure that the number of blocks is valid
+  REQUIRE(MOD(thisPC%A%n,thisPC%numBlocks) .EQ. 0)
+
+  !makes sure that the value of omega is valid
+  REQUIRE(thisPC%omega .LE. 2)
+  REQUIRE(thisPC%omega .GE. 0)
```

Comment:
All of these require statements, I think they need to use the >=, <=, ==, <, and >.